test/staking/SimpleDAIStaking.test.ts

```diff
@@ -1180,6 +1188,55 @@ describe("SimpleDAISTAking - staking with cDAI mocks", () => {
     //   web3.utils.toWei("10", "gwei") + precisionLossDai
     // );
   });
+  it("it should be able stake and withdraw their stake in iToken",async()=>{
+    const stakingAmount = ethers.utils.parseUnits("100",8);
+
+    await cDAI["mint(address,uint256)"](staker.address, stakingAmount);
+    await cDAI.connect(staker).approve(goodCompoundStaking.address, stakingAmount);
+    const productivityBeforeStake = await goodCompoundStaking.getProductivity(staker.address)
+
+    await goodCompoundStaking
+      .connect(staker)
+      .stake(stakingAmount, 100, true)
+      await goodCompoundStaking
+      .connect(staker)
+      .withdrawStake(stakingAmount, true)
+    const productivityAfterWithdraw = await goodCompoundStaking.getProductivity(staker.address)
+    expect(productivityBeforeStake[0]).to.be.equal(productivityAfterWithdraw[0])
+  })
+  it("it should be able to stake in iToken and withdraw in Token",async()=>{
+    const stakingAmount = ethers.utils.parseUnits("100",8);
+    const daiMintAmount = ethers.utils.parseEther("100000")
+    await dai["mint(address,uint256)"](cDAI.address, daiMintAmount);
+    await cDAI["mint(address,uint256)"](staker.address, stakingAmount);
+    await cDAI.connect(staker).approve(goodCompoundStaking.address, stakingAmount);
+    const productivityBeforeStake = await goodCompoundStaking.getProductivity(staker.address)
+    await goodCompoundStaking
+      .connect(staker)
+      .stake(stakingAmount, 100, true)
+      await goodCompoundStaking
+      .connect(staker)
+      .withdrawStake("1603010101010101010101", false) // 1603010101010101010101 is equaliavent of 100cDAI in DAI with currentexchange rate
+    const productivityAfterWithdraw = await goodCompoundStaking.getProductivity(staker.address)
+    expect(productivityBeforeStake[0]).to.be.equal(productivityAfterWithdraw[0])
+  })
+
+  it("it should be able to stake in Token and withdraw in iToken",async()=>{
+    const stakingAmount = ethers.utils.parseEther("100")
+    await dai["mint(address,uint256)"](staker.address, stakingAmount);
+    await dai.connect(staker).approve(goodCompoundStaking.address, stakingAmount);
+    const productivityBeforeStake = await goodCompoundStaking.getProductivity(staker.address)
+    await goodCompoundStaking
+      .connect(staker)
+      .stake(stakingAmount, 100, false)
+    await goodCompoundStaking
+      .connect(staker)
+      .withdrawStake("623826387", true) 
+    await goodCompoundStaking.connect(staker).withdrawStake("000000000036236363637",false);// 000000000036236363637 is precision loss due to itoken decimals < token decimals
+    const productivityAfterWithdraw = await goodCompoundStaking.getProductivity(staker.address)
+    
+    expect(productivityBeforeStake[0]).to.be.equal(productivityAfterWithdraw[0]) 
```

Comment:
test should always check the action expected result.
you stake, you expect user balance to be less and you expect contract balance+records to have changed

Reply:
I added proper variables that test it also